docs/geps/gep-02.rst

```diff
@@ -0,0 +1,158 @@
+| Status     | Draft                                                                   |
++------------+-------------------------------------------------------------------------+
+| Type       | Standards Track                                                         |
++------------+-------------------------------------------------------------------------+
+| Created    | 2020-02-22                                                              |
++------------+-------------------------------------------------------------------------+
+| Resolution | <url> (required for Accepted | Rejected | Withdrawn)                    |
++------------+-------------------------------------------------------------------------+
+
+
+Abstract
+--------
+
+This GEP lays out how GETTSIM stores the user-provided data (be it from the SOEP, EVS,
+example indidviduals...) and passes it around to the functions calculating taxes and
+transfers.
+
+
+Motivation and Scope
+--------------------
+
+This section describes the need for the proposed change. It should describe the existing
+problem, who it affects, what it is trying to solve, and why. This section should
+explicitly address the scope of and key requirements for the proposed change.
+
+
+Usage and Impact
+----------------
+
+* Primarily internal / relevant for developers — highest-level interface can be easily
+  adjusted
+* Affects users only via the interface of lower-level functions (they might want to
+  call those or override them)
+
+
+
+Detailed description
+--------------------
+
+Main idea: adhere to normal forms (see pages https://www.wiwi.uni-bonn.de/gaudecker/_static/prog_econ/2019/09_data_management.pdf).
+
+1. Values do not have any internal structure
+2. Tables do not contain redundant information
+3. No structure in variable names (always use "long" format, not "wide" format)
+
+1. and 3. should be uncontroversial, for 2., there are two potential ways forward, see
+below.
+
+Some things that will be the case for sure:
+
+* Anything variable that exists at a level that is above the lowe
+
+.. note::
+
+    We might get by without steuersubjekt (tax unit) by just tracing out relations
+    between people and eligibility; I leave it in for now.
+
+
+A: Strict adherence to 2nd normal form
+---------------------------------------
+
+* 3+ different tables (bedarfsgemeinschaft, steuersubjekt, person, ?).
+* Indexes would be ``jahr`` :math:`\times` {``bedarfsgemeinschaft``, ``steuersubjekt``,
+  ``person``}
+
+
+B: One table with MultiIndex
+----------------------------
+
+* Index would be ``jahr`` :math:`\times` ``bedarfsgemeinschaft`` :math:`\times`
+  ``steuersubjekt`` :math:`\times` ``person``
+* Any variable relevant at a level higher than ``person`` will be filled in the same
+  way for all persons belonging to that unit.
+
+
+Pro A:
+
+* Very obvious what is the correct level meant
+* Never a need to append anything like ``_hh`` to a variable
```

Comment:
Why not? You'd still might calculate everything on person level first and the sum up.

Reply:
... and by summing up you got rid of the `person` dimension and just use a table that does not include it :-)